### PR TITLE
PowerStore file replication support

### PR DESCRIPTION
**Pull request by ChristianAtDell**
# Description
Started documenting the addition of PowerStore File Replication. 

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| [639](https://github.com/dell/csm/issues/639) |

# Checklist:

- [X] Have you run a grammar and spell checks against your submission?
- [ ] Have you tested the changes locally?
- [X] Have you tested whether the hyperlinks are working properly?
- [ ] Did you add the examples wherever applicable?
- [ ] Have you added high-resolution images?

### CODE REVIEW

**Review by lukeatdell — APPROVED** [review]

File: content/docs/replication/deployment/powerstore.md [hunk 1]
```diff
@@ -172,17 +174,49 @@ parameters:
 
 After creating storage class YAML files, they must be applied to your Kubernetes clusters with `kubectl`.
 
+Creating storage classes for file-type storage is very similar to the process for creating block-type storage classes. You can find sample replication enabled storage class for block storage (Volumes) in the driver repository
```
Comment (santhoshatdell) [inline content/docs/replication/deployment/powerstore.md:177]: Block sample is repeated here. This sample below should be about file type.

File: content/docs/replication/deployment/powerstore.md [hunk 2]
```diff
@@ -172,17 +174,49 @@ parameters:
...
+```yaml
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+  name: "powerstore-replication"
+provisioner: "csi-powerstore.dellemc.com"
+reclaimPolicy: Retain
+volumeBindingMode: Immediate
+parameters:
+  csi.storage.k8s.io/fstype: "ext4"
```
Comment (santhoshatdell) [inline content/docs/replication/deployment/powerstore.md:190]: nfs

File: content/docs/replication/deployment/powerstore.md [hunk 3]
```diff
@@ -172,17 +174,49 @@ parameters:
...
 
+Creating storage classes for file-type storage is very similar to the process for creating block-type storage classes. You can find sample replication enabled storage class for file storage (File Systems) in the driver repository
+[here](https://github.com/dell/csi-powerstore/blob/main/samples/storageclass/powerstore-file-replication.yaml).
+
+It will look like this:
+```yaml
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+  name: "powerstore-replication"
```
Comment (santhoshatdell) [inline content/docs/replication/deployment/powerstore.md:185]: file

File: content/docs/replication/deployment/powerstore.md [hunk 4]
```diff
@@ -172,17 +174,49 @@ parameters:
...
+provisioner: "csi-powerstore.dellemc.com"
+reclaimPolicy: Retain
+volumeBindingMode: Immediate
+parameters:
+  csi.storage.k8s.io/fstype: "ext4"
+  replication.storage.dell.com/nasName: "nas-server"
+  replication.storage.dell.com/allowRoot: "false"
+  replication.storage.dell.com/isReplicationEnabled: "true"
+  replication.storage.dell.com/remoteStorageClassName: "powerstore-replication"
```
Comment (santhoshatdell) [inline content/docs/replication/deployment/powerstore.md:194]: file

**Review by harshaatdell — APPROVED** [review]

**Review by rajkumar-palani — APPROVED** [review]
lgtm

File: content/docs/replication/deployment/powerstore.md [hunk 5]
```diff
@@ -172,17 +174,49 @@ parameters:
...
+
 #### Storage Class Creation With repctl
 
 `repctl` can simplify storage class creation by creating a pair of mirrored storage classes in both clusters
-(using a single storage class configuration) in one command.
+(using a single storage class configuration file) in one command.
 
 To create storage classes with `repctl` you need to fill the config with necessary information.
-You can find an example in [here](https://github.com/dell/csm-replication/blob/main/repctl/examples/powerstore_example_values.yaml), copy it, and modify it to your needs.
+You can find an example for block storage [here](https://github.com/dell/csm-replication/blob/main/repctl/examples/powerstore_block_example_values.yaml) and for file storage [here](https://github.com/dell/csm-replication/blob/main/repctl/examples/powerstore_file_example_values.yaml), copy it, and modify it to your needs.
```
Comment (rajkumar-palani) [inline content/docs/replication/deployment/powerstore.md:215]: can you please check whether link is working?

**Review by rajkumar-palani — COMMENTED** [review]
can you please check the review comments. Thanks for your time.

File: content/docs/replication/deployment/powerstore.md [hunk 6]
```diff
@@ -172,17 +174,49 @@ parameters:
 
 After creating storage class YAML files, they must be applied to your Kubernetes clusters with `kubectl`.
 
+Creating storage classes for file-type storage is very similar to the process for creating block-type storage classes. You can find sample replication enabled storage class for file storage (File Systems) in the driver repository
+[here](https://github.com/dell/csi-powerstore/blob/main/samples/storageclass/powerstore-file-replication.yaml).
```
[the remark this answers is not in the document]
Comment (ChristianAtDell) [reply content/docs/replication/deployment/powerstore.md:178]: The link is to a file that is a part of https://github.com/dell/csi-powerstore/pull/232 and will not work until after that PR has been merged into main. 

For now, all file-replication work for PowerStore is being isolated to a feature branch, and once that work is complete, that feature branch will be merged to main.

File: content/docs/replication/deployment/powerstore.md (hunk 5 above)
Comment (ChristianAtDell) [reply content/docs/replication/deployment/powerstore.md:215]: Same as above. Link is to a file in a PR that will not be merged in until these changes are pulled in as well, which will be when the feature is complete.

**Pull request merged**